src/core/mediator/UnitsHelper.cpp

```diff
@@ -0,0 +1,38 @@
+#include "UnitsHelper.hpp"
+
+ut_system* UnitsHelper::unit_system;
+std::once_flag UnitsHelper::unit_system_inited;
+
+double UnitsHelper::get_converted_value(const std::string &in_units, double value, const std::string &out_units)
+{
+    if(in_units == out_units){
+        return value; // Early-out optimization
+    }
+    if(in_units == "" || out_units == ""){
+        throw std::runtime_error("Unable to process empty units value for pairing \"" + in_units + "\" \"" + out_units + "\"");
+    }
+    std::call_once(unit_system_inited, init_unit_system);
```

Comment:
Why doesn't this init happen in the constructor?

Reply:
That's a static method... at present there is no constructor (is there such a thing as a static constructor in C++?)

Reply:
Hmm...I guess the way it is used you may not need one...but then you may as well just use a namespace (and a nested anonymous namespace) for `private` components.  Unless there is a need for dynamic runtime state, the `class` seems to be the wrong construct here?

Reply:
Dunno. Since this is using a C library in a C++ context it's hard to know how best to adapt to OO... in future we likely may want to cache converters (in addition to the `unit_system`), so there may be more "state" later....and putting them in global variables (even in a namespace) felt old-school (even if they are functionally indistinguishable). Not sure what common/best practice is here. But yes I think there may be more state later. And thread-safety was a concern, I'm not sure if a static class will make that easier or harder than namespaced globals.

Reply:
Fixed (with test) in 7020399.